docs/how-to/debug-in-cluster.md

```diff
@@ -0,0 +1,59 @@
+# Debug a container within a cluster
+
+The container build also publishes a debug container for each tagged release of the container with the tag suffixed with `-debug`. This container contains the workspace and has an alternative entrypoint which allows the devcontainer to attach: if you have configured a `livenessProbe` that requires the service to have started it should be disabled. The container also installs debugpy and makes the service install editable. Any custom `command` or `args` defined for the container should be disabled.
+
+With the [Kubernetes plugin for vscode](https://marketplace.visualstudio.com/items?itemName=ms-kubernetes-tools.vscode-kubernetes-tools) it is then possible to attach to the container inside the cluster. This may require that your targeted kubeconfig is at `~/.kube/config`, rather than referenced from the environment variable `KUBECONFIG`. It may also be necessary to [add additional contextual information](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_config/kubectl_config_set-context/), such as the namespace.
+
+![Location of the Kubernetes plugin in the plugin bar (screen left), with the Clusters>cluster>Workloads>Pods views expanded out to show a pod named "my-service", overlaid with a dropdown box, with "Attach Visual Studio Code" highlighted](../images/debugging-kubernetes.jpg)
+The Kubernetes plugin can be found in the plugin bar. Expanding the Clusters>`cluster`>Workloads>Pods views, your service should be visible. Right Click>Attach Visual Studio Code will initiate connecting to the workspace in the cluster. Select your service container from the top menu when prompted.
+
+After the connection to the cluster has been established, it may be necessary to open the workspace folder by clicking the Explorer option in the plugin bar, the repository will be mounted at `/workspaces/<service name>`, equivalent to when working with a local devcontainer.
```

Comment:
This does appear to always be necessary: container config files are supported for attaching to generic remote devcontainers, but not currently for those in kubernetes.

Reply:
Re kubeconfig. You can select your kubeconfig after launching vscode, see image. The important thing is that kubectl and helm are available on the path. This is challenging when vscode may be launched from gnome (or be already running and attach to the existing process). For this reason I suggest we recommend putting the (current version) tools in the path in .profile like I suggest for docker-compose here https://epics-containers.github.io/main/how-to/compose-quickstart.html#diamond-light-source-workstation. Note that this is peculiar to sites which use 'module load' to get these tools and they are not in the path by default.


![image](https://github.com/user-attachments/assets/def697ae-10d7-4152-9418-0ce52d0274ab)